Sources/KlaviyoSwift/Klaviyo.swift

```diff
@@ -13,6 +13,115 @@ let version = "2.0.0"
+                    self.service = service
+                }
+            }
+            public let metric: Metric
+            public let properties: [String: Any]
+            public let profile: [String: Any]
+            public var time: Date?
+            public let value: Double?
+            public let uniqueId: String
+            public init(metric: Metric,
+                        properties: [String : Any],
+                        profile: [String : Any],
+                        value: Double? = nil,
+                        time: Date? = nil,
+                        uniqueId: String? = nil) {
+                self.profile = profile
+                self.metric = metric
+                self.properties = properties
+                self.value = value
+                self.time = time
+                self.time = time ?? environment.date()
+                self.uniqueId = uniqueId ?? environment.uuid().uuidString
+            }
+            
+        }
+        public let attributes: Attributes
+        public init(attributes: Attributes) {
+            self.attributes = attributes
+        }
+    }
+    
+    public struct Profile {
+        public struct Attributes {
+            public struct Location {
+                public let address1: String?
+                public let address2: String?
+                public let city: String?
+                public let country: String?
+                public let latitude: Double?
+                public let longitude: Double?
+                public let region: String?
+                public let zip: String?
+                public let timezone: String?
+                public init(address1: String?=nil,
+                            address2: String?=nil,
+                            city: String?=nil,
+                            country: String?=nil,
+                            latitude: Double?=nil,
+                            longitude: Double?=nil,
+                            region: String?=nil,
+                            zip: String?=nil,
+                            timezone: String?=TimeZone.autoupdatingCurrent.identifier) {
+                    self.address1 = address1
+                    self.address2 = address2
+                    self.city = city
+                    self.country = country
+                    self.latitude = latitude
+                    self.longitude = longitude
+                    self.region = region
+                    self.zip = zip
+                    self.timezone = timezone
+                }
+            }
+            let email: String?
+            let phoneNumber: String?
+            let externalId: String?
+            let firstName: String?
+            let lastName: String?
+            let organization: String?
+            let title: String?
+            let image: String?
+            let location: Location?
+            let properties: [String: Any]?
+            public init(email: String?=nil,
```

Comment:
is there no data class object that avoids all this boilerplate of init?

Reply:
You get init for free but it's not public.

Reply:
And then if you want optional stuff you need to specify it.